### ISSUE TITLE
slither-disable-next-line do not skip detector

### ISSUE THREAD

**Issue by wpater** [crytic/slither#977]
I have a contract:
```solidity
pragma solidity ^0.8.0;

import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";

contract Abc {
    using EnumerableSet for EnumerableSet.UintSet;
    EnumerableSet.UintSet internal availableOrderLengths;
    constructor (uint256[] memory orderLengths_) {
        for (uint256 i = 0; i < orderLengths_.length; i++) {
            availableOrderLengths.add(orderLengths_[i]);
        }
    }
}

```

Slither is reporting this as `unused-return` for the `availableOrderLengths.add(orderLengths_[i]);` (`ignores return value by availableOrderLengths.add(orderLengths_[i])`.

At this point I am not sure that it is a correct report or false-positive but I tried to use `// slither-disable-next-line unused-return` above this line but this is not working - it is still detected and reported.

Why this `disable` is not working in this case?

**Comment by spilehchiha** [comment]
I used the directive `// slither-disable-next-line unused return` just before the line where the `add` operation happens but inside the body of the loop and it successfully removes the `unused-return` line in the output.
Is it possible that you might have used the directive outside of the body of the loop?
```
pragma solidity ^0.8.0;

import "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";

contract Abc {
    using EnumerableSet for EnumerableSet.UintSet;
    EnumerableSet.UintSet internal availableOrderLengths;
    constructor (uint256[] memory orderLengths_) {
        for (uint256 i = 0; i < orderLengths_.length; i++) {
            // slither-disable-next-line unused-return
            availableOrderLengths.add(orderLengths_[i]);
        }
    }
}
```

**Comment by wpater** [comment]
Yes, exactly in this way, to be sure I copy-pasted your code and - `Abc.constructor(uint256[]) (contracts/temp.sol#8-13) ignores return value by availableOrderLengths.add(orderLengths_[i]) (contracts/temp.sol#11)`.

I am using 0.8.1

**Comment by Jaime-Iglesias** [comment]
@wpater Has this issue been fixed?

**Comment by wpater** [comment]
No, still not working with `--exclude-dependencies` flag.

**Comment by montyly** [comment]
I believe with fixed the issue with #1120, it will be available in the upcoming release
